### PR TITLE
Add prices to testing overhead dashboard

### PR DESCRIPTION
**Pull request by PaliC**
Adds an estimated cost to the testing overhead dashboard (hud.pytorch.org/testing_overhead)

### CODE REVIEW

File: torchci/pages/testing_overhead.tsx [hunk 1]
```diff
@@ -64,6 +64,19 @@ function GenerateOncallTestingOverheadLeaderboard({
...
+              flex: 1,
+              renderCell: (params: GridRenderCellParams<string>) => {
+                  const oncall = params.value;
+                  return (
+                    <a href={`/testing_overhead/oncall_insights?oncall=${oncall}`}>
+                      {oncall}
+                    </a>
+                  );
+                }
+          },
```
Comment (huydhn) [inline torchci/pages/testing_overhead.tsx:79]: Nit: you could run `yarn format` this file to clean up the format for you

File: torchci/pages/testing_overhead.tsx [hunk 2]
```diff
@@ -73,26 +86,29 @@ function GenerateOncallTestingOverheadLeaderboard({
...
                 headerName: "% of workflow time",
                 flex: 1,
                 valueFormatter: (params: GridValueFormatterParams<number>) =>
                   params.value + "%",
                 filterable: false,
               }, 
               {
-                field: "oncall",
-                headerName: "Oncall",
+                field: "estimated_price_per_run_in_dollars",
```
Comment (huydhn) [inline torchci/pages/testing_overhead.tsx:97]: This is ok as the first step, but I think we should figure out a way to show the price = price per run x the number of workflow run.  Mainly to point out the total cost associated with each team / oncall.  Seeing a price per run of $19.23 is only half the picture I think

Comment (huydhn) [reply torchci/pages/testing_overhead.tsx:97]: And showing a bigger bill also has the benefit of nudging people to address it quicker I guess ;)

**Comment by huydhn** [comment]
I have a minor request looking at the preview https://torchci-git-palicprices-fbopensource.vercel.app/testing_overhead.  Could we remove the `Unsharded` part in the table title?  As I have a feeling that it raises a confusing question about that that means for oncall of the team who uses the dashboard

**Comment by huydhn** [comment]
Another UI suggestion is that filtering oncall from the table is a bit hard from my personal experience.  Do you think if it's a good idea to have a dropdown list on top with all the oncalls?  The list could be populated dynamically, i.e. https://github.com/pytorch/test-infra/blob/main/torchci/pages/benchmark/compilers.tsx#L460-L479

**Review by huydhn — APPROVED** [review]
I have some UI comment here and there, but pre-stamp the approval here to unblock you

File: torchci/rockset/commons/__sql/test_time_per_oncall.sql [hunk 3]
```diff
@@ -44,23 +54,45 @@ With avg_table as (
...
+    t.workflow_name as workflow_name, 
+    t.job_name as job_name, 
+    t.test_class as test_class, 
+    t.test_file as test_file, 
+    t.config_job_name as config_job_name, 
+    t.config_shard_name as config_shard_name, 
+    CASE WHEN p.price IS NULL THEN 0 ELSE CAST(p.price AS float) END as price_per_hour 
+  FROM 
+    filtered_table t 
+    LEFT JOIN commons.price_per_config p ON (
```
Comment (huydhn) [inline torchci/rockset/commons/__sql/test_time_per_oncall.sql:71]: Just curious, which process populates this `commons.price_per_config` collection?

**Comment by PaliC** [comment]
Will merge this now, and address the concerns in a separate PR in order to avoid a merge conflict on something else I'm working on.

File: torchci/rockset/commons/__sql/test_time_per_oncall.sql (hunk 3 above)
Comment (PaliC) [reply torchci/rockset/commons/__sql/test_time_per_oncall.sql:71]: I just hard coded it as a csv in rockset. I could make a job to update it nightly though given prices are generally static that may be overkill.

**Pull request merged**